app/code/Magento/CatalogInventory/Model/Indexer/ProductPriceIndexFilter.php

```diff
@@ -104,6 +104,8 @@ public function modifyPrice(IndexTableStructure $priceTable, array $entityIds =
             $select->where('stock_item.use_config_manage_stock = 0 AND stock_item.manage_stock = 1');
         }
 
+        $select->where('stock_item.product_id in (?)', $entityIds);
```

Comment:
I may be wrong but I'm not sure this is what @sidolov intended.

He probably meant that it's not necessary to distinguish between empty or not empty array when building the `$select->where('...IN...', ...)` because if the array is empty this will produce a `WHERE stock_item.product_id in (NULL)` statement that won't return any value.

And that's the point, if you always apply this statement without checking whether the array is empty, an empty array will always produce the above statement and won't return any result, that's not the expected behavior.

I think the code should be:
```
if (count($entityIds)) {
    $select->where('stock_item.product_id in (?)', $entityIds);
}
```